mgwr/summary.py

```diff
@@ -4,27 +4,32 @@
 from .diagnostics import get_AICc
 
 
-def summaryModel(self):
-    summary = '=' * 75 + '\n'
-    summary += "%-54s %20s\n" % ('Model type', self.family.__class__.__name__)
-    summary += "%-60s %14d\n" % ('Number of observations:', self.n)
-    summary += "%-60s %14d\n\n" % ('Number of covariates:', self.k)
+def summaryModel(self) -> str:
+    """Summary of model
+
+    Returns:
+        str: print out the model summary information
+    """
+    summary = f"{'=' * 75} \n"
+    summary += f"{'Model type':<54s} {self.family.__class__.__name__:20s}\n"
+    summary += f"{'Number of observations:':<60s} {self.n:14d}\n"
+    summary += f"{'Number of covariates:':<60s} {self.k:14d}\n\n"
     return summary
 
 
-def summaryGLM(self):
+def summaryGLM(self) -> str:
+    """Summary of GLM
 
-    if self.name_x is not None:
-        XNames = list(self.name_x)
-        if len(XNames) < self.k:
-            XNames = ["Intercept"] + XNames
-    else:
-        XNames = ["X" + str(i) for i in range(self.k)]
+    Returns:
+        str: print out the GLM summary information
+    """
+
+    XNames = [f"X{i}" for i in range(self.k)]
```

Comment:
This is not equivalent of the original code. You are removing functionality here.